tests/rptest/services/compatibility/example_base.py

```diff
@@ -23,7 +23,8 @@ def __init__(self, redpanda):
     # Calls the internal condition and
     # automatically stores the result
     def condition(self, line):
-        self._condition_met = self._condition(line)
+        if not self._condition_met:
+            self._condition_met = self._condition(line)
```

Comment:
This is ugly but necessary for now. This file and all KStreams examples are re-worked in #4461

Reply:
it's cool. ugly stuff happens all the time. it's often useful to add this context directly into the code as a comment because this sort of thing tends to be hard to reverse engineer later when someone else is reading and working on the code. something like a block comment like

```python
        # this is ugly because <...> and to address it we need to <...>
        if not self._condition_met:
            self._condition_met = self._condition(line)
```